core/arch/arm/plat-imx/crypto_conf.mk

```diff
@@ -74,7 +76,10 @@ $(eval $(call cryphw-enable-drv-hw, CMAC))
 
 ifneq ($(filter y, $(CFG_MX6QP) $(CFG_MX6Q) $(CFG_MX6D) $(CFG_MX6DL) \
 	$(CFG_MX6S) $(CFG_MX6SL) $(CFG_MX6SLL) $(CFG_MX6SX)), y)
-$(eval $(call cryphw-enable-drv-hw, RSA))
```

Comment:
> 
> According to what I observed, all of what is implemented in the lib/libnxpse050 can be done like a crypto driver in the core/driver/crypto like done for the caam crypto driver. This will simply a lot all the problem with weak functions and removing libtomcrypt modification.
> We were faced to the same problem when doing the CAAM driver in the sense that we wanted to not touch tomcrypt or mbedtls.
> The core/driver/crypto_api add the flexibility to register a HW driver to do cryptography in place of using SW libs.

yeah what I posted in this PR is a reduced version of the ciphers -I think the se050 library might be a bit more extensive feature wise than CAAM but still the same thing, crypto stuff  (rsa, ecc, hash, hmac, rng, scp03, aes( ctr, cbc, ecb), des (ecb, cbc) des3 (ecb,cbc), huk, T=1 over i2c secured comms,)...still need to add the interfaces to store/retrieve certificates and blobs.

If a driver is the right way to go, sure I can go in that direction. I just need someone to make the call.

Reply:
I prefer to see something like
`
define cryphw-enable-drv-se050-hw
_var := $(strip $(1))
$$(call force, CFG_NXP_SE050_$$(_var)_DRV, y)
$$(call force, CFG_CRYPTO_DRV_$$(_var), y)
endef

ifeq ($(CFG_NXP_SE05X), y)
$(eval $(call cryphw-enable-drv-se050-hw, RSA))
else
$(eval $(call cryphw-enable-drv-hw, RSA))
endif
`

As the SE050 is now a driver, it should be enable like a driver by the platform using it. In plat-imx/conf.mk configuration file.

Reply:
The way I see it we have to distinct types of drivers:
1) drivers that belong to the platform (CAAM)
2) drivers that are common to all platforms (SE050)

Configuring type2 drivers as if they were type1 would lead to a lot of code duplication in each platform makefile.

Shouldn't we better have a central point from where to enable and configure type2 drivers (ie, core/core.mk and core/crypto.mk) and then just adapt the individual platforms to support them?

Reply:
Not sure to follow you on this point. SE050 will not be on all platforms, it will be supported only on platform using an SE050 extension which is custom by definition because this is not supported on all boards.

May be the following macro can be in the common macro file `mk/checkconf.mk`:
`
define cryphw-enable-drv-se050-hw
var := $(strip $(1))
$$(call force, CFG_NXP_SE050$$(_var)DRV, y)
$$(call force, CFG_CRYPTO_DRV$$(_var), y)
endef
`
Then you have to find a generic solution to enable only one crypto driver to be registered to the `crypto_api` to operate a crypto algorithm.
What do you think?

Reply:
yes, those lines make sense to me. going to modify this PR then
I think is necessarily messy since platforms will need to know whether the se050 is enabled before assuming that crypto configs can be forced.